File: tests/fixtures/check_then_act/ts/finally_write_fixture.ts
// A finally block runs on the normal path too, so its write is the guarded write.
export async function finish(repo: any, id: string) {
  const row = repo.get(id);
  if (row) return; // expect: check
  try {
    await repo.sync(id); // expect: gap await
  } finally {
    repo.update(id); // expect: write
  }
}

File: tests/fixtures/check_then_act/py/finally_write_fixture.py
# A finally block runs on the normal path too, so its write is the guarded write.
async def finish(repo, item_id):
    row = repo.get(item_id)
    if row:  # expect: check
        return
    try:
        await repo.sync(item_id)  # expect: gap await
    finally:
        repo.update(item_id)  # expect: write
